BD-2469: Feature Flag experiments (#6267)
braze-inc/braze-docs: _docs/_developer_guide/platform_wide/feature_flags/experiments.md

---
nav_title: Feature Flag Experiments
article_title: Feature Flag Experiments
page_order: 40
description: "Feature flag experiments let you A/B test changes to your applications to optimize conversion rates."
tool: Feature Flags
platform:
  - iOS
  - Android
  - Web

---

# Creating a feature flag experiment

> Feature flag experiments let you A/B test changes to your applications to optimize conversion rates. Marketers can use feature flags to determine whether a new feature positively or negatively impacts conversion rates, or which set of feature flag properties is most optimal.

## Prerequisites

Before you can track user data in the experiment, your app needs to record when a user interacts with a feature flag. This is called a feature flag impression. Make sure to log a feature flag impression whenever a user sees or could have seen the feature you're testing, even if they're in the control group.

> To learn more, see [Logging a feature flag impression][6].

Reviewer: Just double-checking if this should be "Creating feature flags", or something like the below.

```suggestion
To learn more, see [Logging a feature flag impression][6].
```
To learn more about logging feature flag impressions, see [Creating feature flags][6].

```javascript
const featureFlag = braze.getFeatureFlag("my-new-feature");
braze.logFeatureFlagImpression("my-new-feature");
if (featureFlag?.enabled) {
   return <NewFeature />
} else {
   return <ExistingFeature />
}

```

## Step 1: Create an experiment

1. Go to **Messaging** > **Campaigns** and click **+ Create Campaign**.
2. Select **Feature Flag Experiment**.
3. Name your campaign something clear and meaningful.

## Step 2: Add experiment variants

Next, create variations. For each variant, choose the feature flag you want to turn on or off and review the assigned properties.

To test the impact of your feature, use variants to split traffic into two or more groups. One variant should be considered your control group—this variant should have the feature flag turned off.

### Overwriting properties

> While you specified default properties when you originally set up your feature flag, you can choose to overwrite those values for users who receive a specific campaign variant.

Reviewer: ```suggestion
Though you specified default properties when you originally set up your feature flag, you can choose to overwrite those values for users who receive a specific campaign variant.
```

---
nav_title: Feature Flag Experiments
article_title: Feature Flag Experiments
page_order: 40

> description: " "

Author: ```suggestion
description: "Feature flag experiments let you A/B test changes to your applications to optimize conversion rates."
```

tool: Feature Flags
platform:
  - iOS
  - Android
  - Web

---

# Creating a feature flag experiment

> Feature flag experiments let you A/B test changes to your applications to optimize conversion rates. Marketers can use feature flags to determine whether a new feature positively or negatively impacts conversion rates, or which set of feature flag properties is most optimal.

## Prerequisites

Before you can track user data in the experiment, your app needs to record when a user interacts with a feature flag. This is called a feature flag impression. Make sure to log a feature flag impression whenever a user sees or could have seen the feature you're testing, even if they're in the control group.

To learn more, see [Logging a feature flag impression][6].

```javascript
const featureFlag = braze.getFeatureFlag("my-new-feature");
braze.logFeatureFlagImpression("my-new-feature");
if (featureFlag?.enabled) {
   return <NewFeature />
} else {
   return <ExistingFeature />
}

```

## Step 1: Create an experiment

1. Go to **Messaging** > **Campaigns** and click **+ Create Campaign**.
2. Select **Feature Flag Experiment**.
3. Name your campaign something clear and meaningful.

> 4. Add [Teams][1] and [Tags][2] as needed.

Reviewer 2: ```suggestion
```
we don't support it yet

>     - Tags make your campaigns easier to find and build reports out of. For example, when using the [Report Builder][3], you can filter by particular tags.

Reviewer 2: ```suggestion
```

## Step 2: Add experiment variants

Next, create variations. For each variant, choose the feature flag you want to turn on or off and review the assigned properties.

To test the impact of your feature, use variants to split traffic into two or more groups. One variant should be considered your control group—this variant should have the feature flag turned off.

### Overwriting properties

While you specified default properties when you originally set up your feature flag, you can choose to overwrite those values for users who receive a specific campaign variant.

> ![][image1]{: style="max-width:80%"}

Reviewer 2: is this image from canvas? it's missing the on/off switch.

Author: Ah whoops yes, let me recreate this in a campaign

To edit, add, or remove additional default properties, edit the feature flag itself from **Messaging** > **Feature Flags**.

## Step 3: Choose users to target

Next, you need to [target users][4] by choosing segments or filters to narrow down your audience. Segment membership is calculated when feature flags are refreshed for a given user.

{% alert note %}

> Your target audience will be eligible for the feature flag immediately, when your app refreshes feature flags, or when a new session is started.

Reviewer 2: ```suggestion
Your target audience will be eligible for the feature flag as soon as you save a rollout greater than 0%. Changes are made available once your app refreshes feature flags, or when a new session is started.
```
